subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt

```diff
@@ -495,46 +500,68 @@ class RealSubscriptionsManager @Inject constructor(
         planId: String,
         offerId: String?,
         replacementMode: SubscriptionReplacementMode,
+        origin: String?,
     ) = withContext(dispatcherProvider.io()) {
         try {
-            if (!isSignedIn()) {
-                logcat { "Subs: Cannot switch plan - user not signed in" }
-                _currentPurchaseState.emit(CurrentPurchase.Failure("User not signed in for switch"))
-                return@withContext
-            }
-
             val currentSubscription = authRepository.getSubscription()
             if (currentSubscription == null || !currentSubscription.isActive()) {
-                logcat { "Subs: Cannot switch plan - no active subscription found" }
                 _currentPurchaseState.emit(CurrentPurchase.Failure("No active subscription found for switch"))
                 return@withContext
             }
 
+            // Start wide event tracking
+            val isUpgrade = currentSubscription.productId in listOf(MONTHLY_PLAN_US, MONTHLY_PLAN_ROW)
+            val switchType = if (isUpgrade) "upgrade" else "downgrade"
```

Comment:
Perhaps we could move this logic inside `subscriptionSwitchWideEvent.onSwitchFlowStarted()`?